lib/java_buildpack/container/tomee/tomee_resource_configuration.rb

```diff
@@ -84,13 +84,21 @@ def add_relational_resource(service, resources)
       end
 
       def resources_xml
-        @droplet.root + 'WEB-INF/resources.xml'
+        if ear?
+          @droplet.root + 'META-INF/resources.xml'
+        else
+          @droplet.root + 'WEB-INF/resources.xml'
+        end
       end
 
       def read_xml(file)
         File.open(file, 'a+') { |f| REXML::Document.new f }
       end
 
+      def ear?
+        (@application.root + 'META-INF/application.xml').exist?
```

Comment:
Awesome, thanks @violetagg, @kelapure also gave the same feedback. Will check on how else we can see if an exploded app is an ear.